### PR TITLE
Rework LoRA section, fix incongruent package versions required by different samples

### PR DESCRIPTION
**Pull request by yatarkan**

### CODE REVIEW

File: image_generation/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+--extra-index-url https://download.pytorch.org/whl/cpu
+torch==2.2.2+cpu; sys_platform != 'darwin'
+torch==2.2.2; sys_platform == 'darwin'
```
Comment (Wovchena) [inline image_generation/requirements.txt:3]: Please, add a comment explaining the diff

File: image_generation/stable_diffusion_1_5/cpp/README.md [hunk 2]
```diff
@@ -104,13 +116,13 @@ Positive prompt: cyberpunk cityscape like Tokyo New York  with tall buildings at
...
+To read the numpy latent instead of C++ std lib for the alignment with Python pipeline, use `-r, --readNPLatent` argument.
 
 * Generate image without lora `./build/stable_diffusion -r`
 
    ![](./without_lora.bmp)
 
-* Generate image with soulcard lora `./build/stable_diffusion -r`
+* Generate image with soulcard lora `./build/stable_diffusion -r -l path/to/soulcard.safetensors`
 
    ![](./soulcard_lora.bmp)
```
Comment (Wovchena) [inline image_generation/stable_diffusion_1_5/cpp/README.md:128]: While you are nearby, please replace `is differ` to `differ` below.

File: samples/requirements.txt [hunk 3]
```diff
@@ -2,3 +2,4 @@
 optimum[openvino]==1.20.0
 einops==0.8.0  # For Qwen
 transformers_stream_generator==0.0.5  # For Qwen
+-r ../image_generation/requirements.txt
```
Comment (Wovchena) [inline samples/requirements.txt:5]: Reminder to inverse the dependency or introduce a common requirements.txt

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: image_generation/stable_diffusion_1_5/cpp/README.md [hunk 4]
```diff
@@ -52,14 +52,26 @@ python -m pip install ../../../thirdparty/openvino_tokenizers/[transformers]
...
+#### LoRA Weights Format
+
+In this sample LoRA weights are used in [safetensors]((https://huggingface.co/docs/safetensors/index#format)) format.
+Safetensors is a serialization format developed by Hugging Face that is specifically designed for efficiently storing and loading large tensors. It provides a lightweight and efficient way to serialize tensors, making it easier to store and load machine learning models.
+
+The LoRA safetensors model is loaded via [safetensors.h](https://github.com/hsnyder/safetensors.h). The layer name and weight are modified with `Eigen` library and inserted into the SD models with `ov::pass::MatcherPass` in the file [common/diffusers/src/lora.cpp](https://github.com/openvinotoolkit/openvino.genai/blob/master/image_generation/common/diffusers/src/lora.cpp).
+
+#### Using LoRA Weights
+
+SD model [dreamlike-anime-1.0](https://huggingface.co/dreamlike-art/dreamlike-anime-1.0) and LoRA [soulcard model](https://civitai.com/models/67927?modelVersionId=72591) are tested in this pipeline.
```
Comment (Wovchena) [inline image_generation/stable_diffusion_1_5/cpp/README.md:70]: I just realized. No one cares what we test. You can write instead that a good starting point are these two models.

File: image_generation/stable_diffusion_1_5/cpp/README.md (hunk 2 above)
Comment (yatarkan) [reply image_generation/stable_diffusion_1_5/cpp/README.md:128]: Fixed for both SD and LCM

File: samples/requirements.txt (hunk 3 above)
Comment (yatarkan) [reply samples/requirements.txt:5]: Fixed

File: image_generation/stable_diffusion_1_5/cpp/README.md [hunk 5]
```diff
@@ -52,14 +52,26 @@ python -m pip install ../../../thirdparty/openvino_tokenizers/[transformers]
 > [!NOTE]
 > Now the pipeline support batch size = 1 only, i.e. static model `(1, 3, 512, 512)`
 
-### LoRA enabling with safetensors
+### (Optional) Enable LoRA Weights with Safetensors
 
-Refer to [python pipeline blog](https://blog.openvino.ai/blog-posts/enable-lora-weights-with-stable-diffusion-controlnet-pipeline).
-The safetensor model is loaded via [safetensors.h](https://github.com/hsnyder/safetensors.h). The layer name and weight are modified with `Eigen` library and inserted into the SD models with `ov::pass::MatcherPass` in the file [common/diffusers/src/lora.cpp](https://github.com/openvinotoolkit/openvino.genai/blob/master/image_generation/common/diffusers/src/lora.cpp).
+Low-Rank Adaptation(LoRA) is a novel technique introduced to deal with the problem of fine-tuning Diffusers and Large Language Models (LLMs). In the case of Stable Diffusion fine-tuning, LoRA can be applied to the cross-attention layers for the image representations with the latent described.
```
[the remark this answers is not in the document]
Comment (yatarkan) [reply image_generation/stable_diffusion_1_5/cpp/README.md:57]: Fixed

File: image_generation/requirements.txt (hunk 1 above)
Comment (yatarkan) [reply image_generation/requirements.txt:3]: Fixed

File: image_generation/stable_diffusion_1_5/cpp/README.md (hunk 4 above)
Comment (yatarkan) [reply image_generation/stable_diffusion_1_5/cpp/README.md]: Fixed

File: image_generation/stable_diffusion_1_5/cpp/README.md [hunk 6]
```diff
@@ -52,14 +52,26 @@ python -m pip install ../../../thirdparty/openvino_tokenizers/[transformers]
...
+In this sample LoRA weights are used in [safetensors]((https://huggingface.co/docs/safetensors/index#format)) format.
+Safetensors is a serialization format developed by Hugging Face that is specifically designed for efficiently storing and loading large tensors. It provides a lightweight and efficient way to serialize tensors, making it easier to store and load machine learning models.
+
+The LoRA safetensors model is loaded via [safetensors.h](https://github.com/hsnyder/safetensors.h). The layer name and weight are modified with `Eigen` library and inserted into the SD models with `ov::pass::MatcherPass` in the file [common/diffusers/src/lora.cpp](https://github.com/openvinotoolkit/openvino.genai/blob/master/image_generation/common/diffusers/src/lora.cpp).
+
+#### Using LoRA Weights
+
+SD model [dreamlike-anime-1.0](https://huggingface.co/dreamlike-art/dreamlike-anime-1.0) and LoRA [soulcard model](https://civitai.com/models/67927?modelVersionId=72591) are tested in this pipeline.
+
+There are various LoRA models on https://civitai.com/tag/lora and on HuggingFace, you can consider to choose your own LoRA model in safetensor format.
```
[the remark this answers is not in the document]
Comment (yatarkan) [reply image_generation/stable_diffusion_1_5/cpp/README.md]: Fixed, sections are merged

File: image_generation/stable_diffusion_1_5/cpp/README.md [hunk 7]
```diff
@@ -32,7 +32,7 @@ conda env config vars set LD_LIBRARY_PATH=$CONDA_PREFIX/lib:$LD_LIBRARY_PATH
 git submodule update --init
 # Reactivate Conda environment after installing dependencies and setting env vars
 conda activate openvino_sd_cpp
-python -m pip install -r requirements.txt
+python -m pip install -r ../../requirements.txt
 python -m pip install ../../../thirdparty/openvino_tokenizers/[transformers]
 ```
 2. Download a huggingface SD v1.5 model like:
```
[the remark this answers is not in the document]
Comment (yatarkan) [reply image_generation/stable_diffusion_1_5/cpp/README.md]: `runwayml/stable-diffusion-v1-5` and `dreamlike-art/dreamlike-anime-1.0` are Stable Diffusion models exported with `optimum-cli`, not the LoRA models. I suppose it is might be useful to show the user which example models he can download and export.
Reworked this section and made the messaging more clear

**Review by Wovchena — APPROVED** [review]

File: image_generation/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+--extra-index-url https://download.pytorch.org/whl/cpu
+torch==2.2.2+cpu; sys_platform != 'darwin'
+torch==2.2.2; sys_platform == 'darwin'  # Torch wheel for 2.2.2+cpu does not exist for macOS, using 2.2.2 instead
+diffusers==0.27.2
+optimum-intel[openvino]==1.18.0
```
Comment (Wovchena) [inline image_generation/requirements.txt:5]:
```suggestion
```
samples already have `optimum[openvino]==1.20.0`. That's strange that pip didn't detect a conflict

File: image_generation/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,6 @@
+--extra-index-url https://download.pytorch.org/whl/cpu
+torch==2.2.2+cpu; sys_platform != 'darwin'
+torch==2.2.2; sys_platform == 'darwin'  # Torch wheel for 2.2.2+cpu does not exist for macOS, using 2.2.2 instead
```
Comment (Wovchena) [inline image_generation/requirements.txt:3]: I just realized. I'd expect torch to come with optimum. Can torch be removed?

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: image_generation/requirements.txt (hunk 8 above)
Comment (yatarkan) [reply image_generation/requirements.txt]: From samples optimum-intel is installed with version >=1.16.0, so there is no conflict.
But I removed duplicated dependencies from image generation

File: image_generation/requirements.txt (hunk 9 above)
Comment (yatarkan) [reply image_generation/requirements.txt]: Duplicated dependencies removed keeping only diffusers in requirements

**Review by Wovchena — APPROVED** [review]

**Pull request merged**